calico/felix/plugins/fiptgenerator.py

```diff
@@ -551,6 +551,39 @@ def profile_updates(self, profile_id, profile, ip_version, tag_to_ipset,
 
         return updates, deps
 
+    def logged_drop_rules(self, ip_version, chain_name, rule_spec=None,
+                          comment=None, ipt_action="--append", log_pfx=None):
+        """
+        Return a list of iptables updates that can be applied to a chain to
+        drop packets that meet a given rule_spec, with optional log.
+
+        :param ip_version.  Whether these are for the IPv4 or IPv6 iptables.
+        :param chain_name: the chain that the updates will be applied to
+        :param rule_spec: the rule spec (e.g. match criteria).   May be None
+        to drop all packets.
+        :param comment: any comment that should be associated with the
+        rule.  May be None to not include a comment.
+        :param ipt_action: the action that should be used to apply the rule
+        (e.g. --append or --insert)
+        :param log_pfx: If not None, the rules will trigger an iptables LOG
+        action with this log prefix before dropping the packet.
+
+        :return list: a list of iptables fragments of the form
+        [ipt_action] [chain_name] [rule_spec] [action] [comment] e.g.
+        --append my_chain --match conntrack --ctstate INVALID --jump DROP
+        """
+        drop_rules = self.drop_rules(ip_version, chain_name,
+                                     rule_spec=rule_spec, comment=comment,
+                                     ipt_action=ipt_action)
+        if log_pfx is not None:
+            log_rule = " ".join([ipt_action,
+                                 chain_name,
+                                 rule_spec,
+                                 '--jump LOG --log-prefix "%s: " '
+                                 '--log-level 4' % log_pfx])
```

Comment:
This level should be pulled out into a constant.  I also think 4 is a bit too high (particularly since we use this for things that aren't drops).  Maybe 3?

We might need to make it configurable in the rule in future anyway.